<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 CoGAPS | CoGAPS on SciServer</title>
  <meta name="description" content="This introductory course will provide a quick overview of how CoGAPS can be run in RStudio through the SciServer compute platform." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 CoGAPS | CoGAPS on SciServer" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This introductory course will provide a quick overview of how CoGAPS can be run in RStudio through the SciServer compute platform." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 CoGAPS | CoGAPS on SciServer" />
  
  <meta name="twitter:description" content="This introductory course will provide a quick overview of how CoGAPS can be run in RStudio through the SciServer compute platform." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/dasl_favicon.ico" type="image/x-icon" />
<link rel="prev" href="rrstudio.html"/>
<link rel="next" href="about-the-authors.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://practicalgenomics.github.io/cogaps-on-sciserver/"><img src="https://i.imgur.com/l6lpt22.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Course</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#available-course-formats"><i class="fa fa-check"></i><b>0.1</b> Available course formats</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html"><i class="fa fa-check"></i><b>2</b> Getting Started with SciServer</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#what-is-sciserver"><i class="fa fa-check"></i><b>2.1</b> What is SciServer?</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#why-use-sciserver"><i class="fa fa-check"></i><b>2.2</b> Why use SciServer?</a></li>
<li class="chapter" data-level="2.3" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#learning-objectives"><i class="fa fa-check"></i><b>2.3</b> Learning Objectives</a></li>
<li class="chapter" data-level="2.4" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#create-an-account-on-sciserver"><i class="fa fa-check"></i><b>2.4</b> Create an Account on SciServer</a></li>
<li class="chapter" data-level="2.5" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#confirm-your-email-address"><i class="fa fa-check"></i><b>2.5</b> Confirm Your Email Address</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#resources"><i class="fa fa-check"></i><b>2.5.1</b> Resources</a></li>
<li class="chapter" data-level="2.5.2" data-path="getting-started-with-sciserver.html"><a href="getting-started-with-sciserver.html#references"><i class="fa fa-check"></i><b>2.5.2</b> References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cellxgene.html"><a href="cellxgene.html"><i class="fa fa-check"></i><b>3</b> cellxgene</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cellxgene.html"><a href="cellxgene.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="3.2" data-path="cellxgene.html"><a href="cellxgene.html#instructions"><i class="fa fa-check"></i><b>3.2</b> Instructions</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="cellxgene.html"><a href="cellxgene.html#confirm-access"><i class="fa fa-check"></i><b>3.2.1</b> Confirm Access</a></li>
<li class="chapter" data-level="3.2.2" data-path="cellxgene.html"><a href="cellxgene.html#join-genomics-domain"><i class="fa fa-check"></i><b>3.2.2</b> Join Genomics Domain</a></li>
<li class="chapter" data-level="3.2.3" data-path="cellxgene.html"><a href="cellxgene.html#launch-cellxgene"><i class="fa fa-check"></i><b>3.2.3</b> Launch cellxgene</a></li>
<li class="chapter" data-level="3.2.4" data-path="cellxgene.html"><a href="cellxgene.html#edit-dashboard"><i class="fa fa-check"></i><b>3.2.4</b> Edit Dashboard</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="rrstudio.html"><a href="rrstudio.html"><i class="fa fa-check"></i><b>4</b> R/RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rrstudio.html"><a href="rrstudio.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="4.2" data-path="rrstudio.html"><a href="rrstudio.html#introduction-1"><i class="fa fa-check"></i><b>4.2</b> Introduction</a></li>
<li class="chapter" data-level="4.3" data-path="rrstudio.html"><a href="rrstudio.html#instructions-1"><i class="fa fa-check"></i><b>4.3</b> Instructions</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="rrstudio.html"><a href="rrstudio.html#start-up-a-c-moor-rstudio-compute-container"><i class="fa fa-check"></i><b>4.3.1</b> Start up a “C-MOOR RStudio” compute container</a></li>
<li class="chapter" data-level="4.3.2" data-path="rrstudio.html"><a href="rrstudio.html#complete-your-first-swirl-tutorial"><i class="fa fa-check"></i><b>4.3.2</b> Complete your first swirl tutorial</a></li>
<li class="chapter" data-level="4.3.3" data-path="rrstudio.html"><a href="rrstudio.html#managing-your-c-moor-rstudio-compute-container"><i class="fa fa-check"></i><b>4.3.3</b> Managing your C-MOOR RStudio compute container</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cogaps.html"><a href="cogaps.html"><i class="fa fa-check"></i><b>5</b> CoGAPS</a>
<ul>
<li class="chapter" data-level="5.1" data-path="cogaps.html"><a href="cogaps.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="5.2" data-path="cogaps.html"><a href="cogaps.html#what-is-cogaps"><i class="fa fa-check"></i><b>5.2</b> What is CoGAPS?</a></li>
<li class="chapter" data-level="5.3" data-path="cogaps.html"><a href="cogaps.html#what-is-cogaps-used-for"><i class="fa fa-check"></i><b>5.3</b> What is CoGAPS used for?</a></li>
<li class="chapter" data-level="5.4" data-path="cogaps.html"><a href="cogaps.html#instructions-2"><i class="fa fa-check"></i><b>5.4</b> Instructions</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="cogaps.html"><a href="cogaps.html#start-up-a-cogaps-rstudio-compute-container"><i class="fa fa-check"></i><b>5.4.1</b> Start up a “CoGAPS RStudio” compute container</a></li>
<li class="chapter" data-level="5.4.2" data-path="cogaps.html"><a href="cogaps.html#update-rlang"><i class="fa fa-check"></i><b>5.4.2</b> Update rlang</a></li>
<li class="chapter" data-level="5.4.3" data-path="cogaps.html"><a href="cogaps.html#install-packages"><i class="fa fa-check"></i><b>5.4.3</b> Install Packages</a></li>
<li class="chapter" data-level="5.4.4" data-path="cogaps.html"><a href="cogaps.html#load-packages"><i class="fa fa-check"></i><b>5.4.4</b> Load Packages</a></li>
<li class="chapter" data-level="5.4.5" data-path="cogaps.html"><a href="cogaps.html#load-data"><i class="fa fa-check"></i><b>5.4.5</b> Load Data</a></li>
<li class="chapter" data-level="5.4.6" data-path="cogaps.html"><a href="cogaps.html#configure-cogaps"><i class="fa fa-check"></i><b>5.4.6</b> Configure CoGAPS</a></li>
<li class="chapter" data-level="5.4.7" data-path="cogaps.html"><a href="cogaps.html#run-cogaps"><i class="fa fa-check"></i><b>5.4.7</b> Run CoGAPS</a></li>
<li class="chapter" data-level="5.4.8" data-path="cogaps.html"><a href="cogaps.html#visualize-patterns"><i class="fa fa-check"></i><b>5.4.8</b> Visualize Patterns</a></li>
<li class="chapter" data-level="5.4.9" data-path="cogaps.html"><a href="cogaps.html#find-pattern-markers"><i class="fa fa-check"></i><b>5.4.9</b> Find Pattern Markers</a></li>
<li class="chapter" data-level="5.4.10" data-path="cogaps.html"><a href="cogaps.html#document-software"><i class="fa fa-check"></i><b>5.4.10</b> Document Software</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="cogaps.html"><a href="cogaps.html#resources-1"><i class="fa fa-check"></i><b>5.5</b> Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="6" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i><b>6</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by:</a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CoGAPS on SciServer</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=1.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/style.css" rel="stylesheet">
</head>



<div class="hero-image-container">
  <img class= "hero-image" src= "https://github.com/jhudsl/OTTR_Template/raw/main/assets/dasl_thin_main_image.png">
</div>
<div id="cogaps" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> CoGAPS</h1>
<div id="learning-objectives-3" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Learning Objectives</h2>
<ul>
<li>Learn about CoGAPS</li>
<li>How to load packages in RStudio</li>
<li>How to load data in RStudio</li>
<li>How to configure CoGAPS</li>
<li>How to run CoGAPS</li>
<li>How to visualize patterns</li>
<li>How to find pattern markers</li>
<li>How to document software</li>
</ul>
</div>
<div id="what-is-cogaps" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> What is CoGAPS?</h2>
<p>CoGAPS (Coordinated Gene Activity across Pattern Subsets) is a Bayesian NMF (Nonnegative Matrix Factorization) algorithm. It can be used to perform sparse matrix factorization on any data, and when this data represents biomolecules, to do gene set analysis. CoGAPS improves on other enrichment measurement methods by combining a Markov chain Monte Carlo (MCMC) matrix factorization algorithm (GAPS) with a threshold-independent statistic inferring activity on gene sets.</p>
</div>
<div id="what-is-cogaps-used-for" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> What is CoGAPS used for?</h2>
<p>CoGAPS can be used to perform sparse matrix factorization on any data. And when this data represents biomolecules, to do gene set analysis.</p>
</div>
<div id="instructions-2" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Instructions</h2>
<div id="start-up-a-cogaps-rstudio-compute-container" class="section level3" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> Start up a “CoGAPS RStudio” compute container</h3>
<ol style="list-style-type: decimal">
<li>Open <a href="https://sciserver.org">sciserver.org</a> in a web browser and log in to your account.</li>
</ol>
<p>      a. If you’re already logged in, click “<strong>Home</strong>” in the top menu bar to return to the home page.</p>
<ol start="2" style="list-style-type: decimal">
<li><p>Scroll down to the second set of boxes and click “<strong>Compute</strong>”.</p></li>
<li><p>Click “<strong>Create container</strong>”</p></li>
</ol>
<p>      a. Give your container a name. This can be anything you like, but it’s useful if it says something about the purpose of the container so that you can tell your containers apart. You could name this container “CoGAPS”, since you’ll be using it to run CoGAPS.</p>
<p>      b. In the “<strong>Compute Image</strong>” drop-down menu, select “<strong>R [version #] (RStudio)</strong>” (there may be multiple versions of R in this drop-down menu; click on the latest version listed).</p>
<p>      c. Click “<strong>Create</strong>”. This may take a moment.</p>
<ol start="4" style="list-style-type: decimal">
<li>You should now see a new entry in your list of containers.</li>
</ol>
<p>      a. “Created at” should be a few moments ago.</p>
<p>      b. “Name” should be the name you chose.</p>
<p>      c. “Image” should be <code>"R[version #](RStudio)"</code></p>
<ol start="5" style="list-style-type: decimal">
<li>Start your CoGAPS RStudio container by clicking on its <strong>name</strong> (whatever name you chose when you created the container). This will open in a new tab.</li>
</ol>
<p>      a. You should see RStudio.</p>
<p><img src="resources/images/05-CoGAPS_files/figure-html/16jh1ov1PyRyPKMTJ7ROiEyNm1B5KxdQlYQovVBCYesk_g25954290abf_0_12.png" title="Image of RStudio screen" alt="Image of RStudio screen" width="100%" /></p>
<p>      b. If you see something else, you may have picked the wrong “Compute Image” from the drop-down menu.</p>
<p><strong>If anything goes wrong, you can always delete your container by clicking the red “X” in the last column, and create a new container.</strong></p>
</div>
<div id="update-rlang" class="section level3" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> Update rlang</h3>
<ol style="list-style-type: decimal">
<li>Enter and run these commands into RStudio to make sure that you have the current version of <code>rlang</code> (you will receive an error later on otherwise):</li>
</ol>
<pre><code>packageVersion(&quot;rlang&quot;) # ‘1.0.6’
devtools::install_github(&quot;r-lib/rlang&quot;)
packageVersion(&quot;rlang&quot;) # ‘1.1.0.9000’</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Once you see the output: <code>[1] '1.1.1.9000'</code> rlang has been updated.</li>
</ol>
</div>
<div id="install-packages" class="section level3" number="5.4.3">
<h3><span class="header-section-number">5.4.3</span> Install Packages</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>devtools::install_github(&quot;FertigLab/CoGAPS&quot;)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li><p>This <u>will</u> take a while. Once the red “STOP” symbol is no longer visible in the top right hand corner of the Console, the installation is complete.</p></li>
<li><p>Enter and run this command:</p></li>
</ol>
<pre><code>devtools::install_github(&quot;sjmgarnier/viridis&quot;)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li><p>Once the red “STOP” symbol is no longer visible in the top right hand corner of the Console, the installation is complete.</p></li>
<li><p>Enter and run this command:</p></li>
</ol>
<pre><code>remotes::install_github(&quot;satijalab/seurat&quot;, &quot;seurat5&quot;, quiet = TRUE)</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>This <u>will</u> take a while. Once the red “STOP” symbol is no longer visible in the top right hand corner of the Console, the installation is complete.</li>
</ol>
</div>
<div id="load-packages" class="section level3" number="5.4.4">
<h3><span class="header-section-number">5.4.4</span> Load Packages</h3>
<ol style="list-style-type: decimal">
<li>Enter and run these commands:</li>
</ol>
<pre><code>library( &quot;CoGAPS&quot; )
library( &quot;Seurat&quot; )
library( &quot;viridis&quot; )</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Once the red “STOP” symbol is no longer visible in the top right hand corner of the Console, the packages have been loaded.</li>
</ol>
</div>
<div id="load-data" class="section level3" number="5.4.5">
<h3><span class="header-section-number">5.4.5</span> Load Data</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>url &lt;- &quot;https://github.com/FertigLab/CoGAPS/raw/master/data/inputdata.Rds&quot;
download.file( url, &quot;inputdata.Rds&quot; )</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>You should see this output if the run is successful:</li>
</ol>
<pre><code>trying URL &#39;https://github.com/FertigLab/CoGAPS/raw/master/data/inputdata.Rds&#39;
Content type &#39;application/octet-stream&#39; length 433262849 bytes (413.2 MB)
==================================================
downloaded 413.2 MB</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>pdac_data &lt;- readRDS( &quot;inputdata.Rds&quot; )
pdac_data</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>You should see this output if the run is successful:</li>
</ol>
<pre><code>An object of class Seurat 
15184 features across 25442 samples within 2 assays 
Active assay: originalexp (15176 features, 2000 variable features)
 1 other assay present: CoGAPS
 5 dimensional reductions calculated: PCA, Aligned, UMAP, pca, umap</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>pdac_epi_counts &lt;- as.matrix( pdac_data@assays$originalexp@counts )</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>You should see this output if the run is successful:</li>
</ol>
<pre><code>Warning: sparse-&gt;dense coercion: allocating vector of size 2.9 GiB</code></pre>
</div>
<div id="configure-cogaps" class="section level3" number="5.4.6">
<h3><span class="header-section-number">5.4.6</span> Configure CoGAPS</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>pdac_params &lt;- CogapsParams(
  nIterations=100,          # run for 100 iterations 
  seed=42,                  # for consistency across stochastic runs
  nPatterns=8,              # each thread will learn 8 patterns
  sparseOptimization=TRUE,  # optimize for sparse data
  distributed=&quot;genome-wide&quot; # parallelize across sets
)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>pdac_params &lt;- setDistributedParams( pdac_params, nSets=7 )</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>You should see this output if the run is successful:</li>
</ol>
<pre><code>setting distributed parameters - call this again if you change nPatterns</code></pre>
</div>
<div id="run-cogaps" class="section level3" number="5.4.7">
<h3><span class="header-section-number">5.4.7</span> Run CoGAPS</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>Sys.time()</code></pre>
<ol start="2" style="list-style-type: decimal">
<li><p>Your output should include today’s date and time.</p></li>
<li><p>Enter and run this command:</p></li>
</ol>
<pre><code>pdac_epi_result &lt;- CoGAPS( pdac_epi_counts, pdac_params )</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>You should see this output if the run is successful (This <u>will</u> take a while, most likely around 20+ minutes):</li>
</ol>
<pre><code>This is CoGAPS version 3.19.1 
Running genome-wide CoGAPS on pdac_epi_counts (15176 genes and 25442 samples) with parameters:

-- Standard Parameters --
nPatterns            8 
nIterations          100 
seed                 42 
sparseOptimization   TRUE 
distributed          genome-wide 

-- Sparsity Parameters --
alpha          0.01 
maxGibbsMass   100 

-- Distributed CoGAPS Parameters -- 
nSets          7 
cut            8 
minNS          4 
maxNS          11 

Creating subsets...
set sizes (min, mean, max): (2168, 2168, 2168)
Running Across Subsets...

Data Model: Sparse, Normal
Sampler Type: Sequential
Loading Data...
Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
Done! (00:00:36)
    worker 1 is starting!
-- Equilibration Phase --

Warning: Large values detected, is data log transformed?
    worker 6 is starting!

Warning: Large values detected, is data log transformed?
    worker 4 is starting!

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
    worker 5 is starting!

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
    worker 3 is starting!

Warning: Large values detected, is data log transformed?
    worker 7 is starting!

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
    worker 2 is starting!
-- Sampling Phase --
    worker 2 is finished! Time: 00:18:04
    worker 7 is finished! Time: 00:18:20
    worker 4 is finished! Time: 00:18:49
    worker 3 is finished! Time: 00:18:42
    worker 6 is finished! Time: 00:18:59
    worker 1 is finished! Time: 00:19:09
    worker 5 is finished! Time: 00:19:19

Matching Patterns Across Subsets…
Running Final Stage...

Connected to your session in progress, last started 2023-Jun-29 16:26:07 UTC (3 hours ago)
Data Model: Sparse, Normal
Sampler Type: Sequential
Loading Data...
Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
Done! (00:00:29)
    worker 1 is starting!
-- Equilibration Phase --

Warning: Large values detected, is data log transformed?
    worker 5 is starting!

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
    worker 4 is starting!

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
    worker 6 is starting!

Warning: Large values detected, is data log transformed?
    worker 2 is starting!

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?

Warning: Large values detected, is data log transformed?
    worker 3 is starting!

Warning: Large values detected, is data log transformed?
    worker 7 is starting!
-- Sampling Phase --
   worker 1 is finished! Time: 00:17:04
    worker 6 is finished! Time: 00:17:04
    worker 5 is finished! Time: 00:17:16
    worker 4 is finished! Time: 00:17:15
    worker 7 is finished! Time: 00:17:07
    worker 2 is finished! Time: 00:17:17
    worker 3 is finished! Time: 00:17:11</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>Sys.time()</code></pre>
<ol start="6" style="list-style-type: decimal">
<li><p>Your output should include today’s date and time.</p></li>
<li><p>Enter and run this command:</p></li>
</ol>
<pre><code>saveRDS( pdac_epi_result, &quot;../data/pdac_epi_cogaps_result&quot; )</code></pre>
<ol start="8" style="list-style-type: decimal">
<li>You should see this output if the run is successful:</li>
</ol>
<pre><code>Warning: cannot open compressed file &#39;../data/pdac_epi_cogaps_result&#39;, probable reason &#39;No such file or directory&#39;Error in gzfile(file, mode) : cannot open the connection</code></pre>
<ol start="9" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>pdac_epi_result</code></pre>
<ol start="10" style="list-style-type: decimal">
<li>You should see this output if the run is successful:</li>
</ol>
<pre><code>[1] &quot;CogapsResult object with 15176 features and 25442 samples&quot;
[1] &quot;7 patterns were learned&quot;</code></pre>
<ol start="11" style="list-style-type: decimal">
<li>Enter and run this command in order to save your results:</li>
</ol>
<pre><code>saveRDS( pdac_epi_result, &quot;pdac_epi_cogaps_result.rds&quot; )</code></pre>
</div>
<div id="visualize-patterns" class="section level3" number="5.4.8">
<h3><span class="header-section-number">5.4.8</span> Visualize Patterns</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>cogapsresult &lt;- readRDS( &quot;pdac_epi_cogaps_result.rds&quot; )</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>patterns_in_order &lt;- t( cogapsresult@sampleFactors[colnames(pdac_data),] )
pdac_data[[&quot;CoGAPS&quot;]] &lt;- CreateAssayObject( counts = patterns_in_order )</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>You should receive this output:</li>
</ol>
<pre><code>Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes (&#39;-&#39;)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>inputdata &lt;- pdac_data
DefaultAssay(inputdata) &lt;- &quot;CoGAPS&quot;
pattern_names = rownames( inputdata@assays$CoGAPS )</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>color_palette &lt;- viridis(n=10)
FeaturePlot(inputdata, pattern_names, cols=color_palette, reduction = &quot;umap&quot;) &amp; NoLegend()</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Your output should look like this if the run was successful (visible in the bottom right corner of your screen):</li>
</ol>
<p><img src="resources/images/05-CoGAPS_files/figure-html/16jh1ov1PyRyPKMTJ7ROiEyNm1B5KxdQlYQovVBCYesk_g25a5f9f2e31_0_1.png" title="Image of a CoGAPS feature plot" alt="Image of a CoGAPS feature plot" width="100%" /></p>
</div>
<div id="find-pattern-markers" class="section level3" number="5.4.9">
<h3><span class="header-section-number">5.4.9</span> Find Pattern Markers</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>pm &lt;- patternMarkers( cogapsresult, threshold=&quot;cut&quot; )</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>You should see this output if the run was successful:</li>
</ol>
<pre><code>Warning: STATS is longer than the extent of &#39;dim(x)[MARGIN]&#39;</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code># hallmarks &lt;- PatternHallmarks( cogapsresult )</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code># plotPatternHallmarks(hallmarks, whichpattern = 7)</code></pre>
</div>
<div id="document-software" class="section level3" number="5.4.10">
<h3><span class="header-section-number">5.4.10</span> Document Software</h3>
<ol style="list-style-type: decimal">
<li>Enter and run this command:</li>
</ol>
<pre><code>sessionInfo()</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Your output should include all the information about your RStudio session.</li>
</ol>
<p><strong>Troubleshooting/Reminders:</strong></p>
<ul>
<li>If you restart RStudio, you must repeat the <a href="https://practicalgenomics.github.io/cogaps-on-sciserver/cogaps.html#load-packages">Load Packages</a> step, otherwise errors will occur and you will not be able to successfully run your code.</li>
</ul>
</div>
</div>
<div id="resources-1" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> Resources</h2>
<p><a href="https://fertiglab.github.io/CoGAPSGuide/">CoGAPS Guide Website</a></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="cogaps.html#cb42-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.2 (2020-06-22)
##  os       Ubuntu 20.04.5 LTS          
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2023-07-20                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package     * version date       lib source                            
##  assertthat    0.2.1   2019-03-21 [1] RSPM (R 4.0.5)                    
##  bookdown      0.24    2023-03-28 [1] Github (rstudio/bookdown@88bc4ea) 
##  bslib         0.4.2   2022-12-16 [1] CRAN (R 4.0.2)                    
##  cachem        1.0.7   2023-02-24 [1] CRAN (R 4.0.2)                    
##  callr         3.5.0   2020-10-08 [1] RSPM (R 4.0.2)                    
##  cli           3.6.1   2023-03-23 [1] CRAN (R 4.0.2)                    
##  crayon        1.3.4   2017-09-16 [1] RSPM (R 4.0.0)                    
##  curl          4.3     2019-12-02 [1] RSPM (R 4.0.3)                    
##  desc          1.2.0   2018-05-01 [1] RSPM (R 4.0.3)                    
##  devtools      2.3.2   2020-09-18 [1] RSPM (R 4.0.3)                    
##  digest        0.6.25  2020-02-23 [1] RSPM (R 4.0.0)                    
##  ellipsis      0.3.1   2020-05-15 [1] RSPM (R 4.0.3)                    
##  evaluate      0.20    2023-01-17 [1] CRAN (R 4.0.2)                    
##  fansi         0.4.1   2020-01-08 [1] RSPM (R 4.0.0)                    
##  fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.0.2)                    
##  fs            1.5.0   2020-07-31 [1] RSPM (R 4.0.3)                    
##  glue          1.4.2   2020-08-27 [1] RSPM (R 4.0.5)                    
##  highr         0.8     2019-03-20 [1] RSPM (R 4.0.3)                    
##  hms           0.5.3   2020-01-08 [1] RSPM (R 4.0.0)                    
##  htmltools     0.5.5   2023-03-23 [1] CRAN (R 4.0.2)                    
##  httr          1.4.2   2020-07-20 [1] RSPM (R 4.0.3)                    
##  jquerylib     0.1.4   2021-04-26 [1] CRAN (R 4.0.2)                    
##  jsonlite      1.7.1   2020-09-07 [1] RSPM (R 4.0.2)                    
##  knitr         1.33    2023-03-28 [1] Github (yihui/knitr@a1052d1)      
##  lifecycle     1.0.3   2022-10-07 [1] CRAN (R 4.0.2)                    
##  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.0.2)                    
##  memoise       2.0.1   2021-11-26 [1] CRAN (R 4.0.2)                    
##  ottrpal       1.0.1   2023-03-28 [1] Github (jhudsl/ottrpal@151e412)   
##  pillar        1.9.0   2023-03-22 [1] CRAN (R 4.0.2)                    
##  pkgbuild      1.1.0   2020-07-13 [1] RSPM (R 4.0.2)                    
##  pkgconfig     2.0.3   2019-09-22 [1] RSPM (R 4.0.3)                    
##  pkgload       1.1.0   2020-05-29 [1] RSPM (R 4.0.3)                    
##  prettyunits   1.1.1   2020-01-24 [1] RSPM (R 4.0.3)                    
##  processx      3.4.4   2020-09-03 [1] RSPM (R 4.0.2)                    
##  ps            1.4.0   2020-10-07 [1] RSPM (R 4.0.2)                    
##  R6            2.4.1   2019-11-12 [1] RSPM (R 4.0.0)                    
##  readr         1.4.0   2020-10-05 [1] RSPM (R 4.0.2)                    
##  remotes       2.2.0   2020-07-21 [1] RSPM (R 4.0.3)                    
##  rlang         1.1.0   2023-03-14 [1] CRAN (R 4.0.2)                    
##  rmarkdown     2.10    2023-03-28 [1] Github (rstudio/rmarkdown@02d3c25)
##  rprojroot     2.0.3   2022-04-02 [1] CRAN (R 4.0.2)                    
##  sass          0.4.5   2023-01-24 [1] CRAN (R 4.0.2)                    
##  sessioninfo   1.1.1   2018-11-05 [1] RSPM (R 4.0.3)                    
##  stringi       1.5.3   2020-09-09 [1] RSPM (R 4.0.3)                    
##  stringr       1.4.0   2019-02-10 [1] RSPM (R 4.0.3)                    
##  testthat      3.0.1   2023-03-28 [1] Github (R-lib/testthat@e99155a)   
##  tibble        3.2.1   2023-03-20 [1] CRAN (R 4.0.2)                    
##  usethis       1.6.3   2020-09-17 [1] RSPM (R 4.0.2)                    
##  utf8          1.1.4   2018-05-24 [1] RSPM (R 4.0.3)                    
##  vctrs         0.6.1   2023-03-22 [1] CRAN (R 4.0.2)                    
##  withr         2.3.0   2020-09-22 [1] RSPM (R 4.0.2)                    
##  xfun          0.26    2023-03-28 [1] Github (yihui/xfun@74c2a66)       
##  yaml          2.2.1   2020-02-01 [1] RSPM (R 4.0.3)                    
## 
## [1] /usr/local/lib/R/site-library
## [2] /usr/local/lib/R/library</code></pre>

</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="rrstudio.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="about-the-authors.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
